test_plans/reduced_feature_set.asciidoc

```diff
@@ -0,0 +1,45 @@
+:sectnums:
+:xrefstyle: short
+
+= Test plan for reduced and full feature set
+
+This is a test plan for the full and reduced feature sets as described in Appendix B.1. and Appendix B.2. of the SYCL 2020 specification.
+
+Currently, the CTS does not take into account the full and reduced feature sets. This test plan proposes a way to modify the CTS in such a way that it may be specified whether conformance should be tested for the reduced or the full feature set. For the remainder of this plan, let "full feature" refer to a feature that is part of the full feature set, but not part of the reduced feature set.
+
+A CTS option is introduced, `SYCL_CTS_ENABLE_FEATURE_SET_FULL`, which specifies that full features should be tested in addition to the features of the reduced feature set. The tests for the full features should be guarded if `#ifdef SYCL_CTS_ENABLE_FEATURE_SET_FULL`.
+
+In all existing tests, the full features are removed and replaced with a workaround. Since the CTS was not written with the feature sets in mind, a "best effort" is made to remove the full features in the CTS. New additions should not use full features unless guarded with `SYCL_CTS_ENABLE_FEATURE_SET_FULL`.
+
+An exception to this are the tests that explicitly test full features. For example, when launching a kernel to test `sycl::id` 's interface, the kernel should be launched with a kernel name instead of using the full feature to not specify a name. However, the test that verifies that the full feature that allows for kernels to be launched without a kernel name works correctly, should still exist and be guarded with `SYCL_CTS_ENABLE_FEATURE_SET_FULL`.
+
+Currently, three full features are defined. For each of these features, the required behavior as a result from removing full features is stated, in addition to the tests which are the exceptions (or the proposal for a simple test if none exist).
+
+== Un-named SYCL kernel functions
+=== Required behavior
+Every kernel launch has a kernel name, e.g. `parallel_for<kernel_name>`.
```

Comment:
I think this is a good goal.  I wonder how much work it will be, though.  Adding a unique kernel name might be difficult in tests where the kernel is defined as part of a complex template.

Reply:
From what I can tell, a good chunk of the tests already use a kernel name. Creating a name for a kernel that is part of a complex template will not be that hard since kernel names themselves can also be templatized e.g. [buffer_constructors_common.h:L67](https://github.com/KhronosGroup/SYCL-CTS/blob/SYCL-2020/tests/buffer/buffer_constructors_common.h#L67).